protocol/httphelper.go

```diff
@@ -115,6 +119,58 @@ func DoRawHTTPRequest(rhost string, rport int, uri string, verb string) bool {
 	return success
 }
 
+// Cache the respsone in the database for later reuse.
+func cacheResponse(req *http.Request, resp *http.Response) {
+	parsedURL, _ := url.Parse(req.URL.String())
+	port, err := strconv.Atoi(parsedURL.Port())
+	if err != nil {
+		output.PrintFrameworkError(err.Error())
+
+		return
+	}
+
+	respBuffer := &bytes.Buffer{}
+	err = resp.Write(respBuffer)
+	if err != nil {
+		output.PrintfFrameworkError("Resp write error: %s", err.Error())
+
+		return
+	}
+
+	db.CacheHTTPResponse(parsedURL.Hostname(), port, parsedURL.Path, respBuffer.String())
+}
+
+// Look up matching URI in the HTTP cache and return it if found.
+func cacheLookup(uri string) (*http.Response, string, bool) {
```

Comment:
Should this be keying on more than just the URI? I could see situations where you have multiple HTTP verbs to the same endpoint leading to incorrect cache hits

Reply:
Caching only supports the HTTP GET verb (see the usage in HTTP).